src/Game/UI/Controls/StbTextBoxCombo.cs

```diff
@@ -30,178 +30,178 @@
+            {
+                _selectedIndex = value;
+
+                if (_items != null)
+                {
+                    UIManager.GetGump<TextboxComboContextMenu>()?.Dispose();
+                    Text = _items[_selectedIndex];
+                    CaretIndex = Text.Length;
+                    SetKeyboardFocus();
+                    OnOptionSelected?.Invoke(this, value);
+                }
+            }
+        }
+
+
+        protected override void OnMouseUp(int x, int y, MouseButtonType button)
+        {
+            if (button != MouseButtonType.Left)
+            {
+                return;
+            }
+
+            if (_arrow != null)
+            {
+                _arrow.Graphic = ARROW_UP;
+                if (x > _arrow.X && x < _arrow.X + _arrow.Width && y > _arrow.Y && y < _arrow.Y + _arrow.Height)
+                {
+                    OnBeforeContextMenu?.Invoke(this, null);
+                    TextboxComboContextMenu contextMenu = new TextboxComboContextMenu(_items, Width - 9, 100, _font, selectedIndex => SelectedIndex = selectedIndex)
+                    {
+                        X = ScreenCoordinateX - 6,
+                        Y = ScreenCoordinateY + Height + 5
+                    };
+                    if (contextMenu.Height + ScreenCoordinateY > Client.Game.Window.ClientBounds.Height)
+                    {
+                        contextMenu.Y -= contextMenu.Height + ScreenCoordinateY - Client.Game.Window.ClientBounds.Height;
+                    }
+
+                    UIManager.Add(contextMenu);
+                }
+            }
+
+            base.OnMouseUp(x, y, button);
+        }
+        protected override void OnMouseDown(int x, int y, MouseButtonType button)
+        {
+            if (button != MouseButtonType.Left)
+            {
+                return;
+            }
+
+            if (_arrow != null && x > _arrow.X && x < _arrow.X + _arrow.Width && y > _arrow.Y && y < _arrow.Y + _arrow.Height)
+            {
+                _arrow.Graphic = ARROW_DOWN;
+            }
+
+            base.OnMouseDown(x, y, button);
+        }
+
+        internal string GetSelectedItem => Text;
+
+        internal uint GetItemsLength => (uint)_items.Length;
+
+        public event EventHandler<int> OnOptionSelected;
+        public event EventHandler OnBeforeContextMenu;
+    }
+
+
+    internal class TextboxComboContextMenu : Control
+    {
+        private readonly Action<int> _setIndex;
+        public TextboxComboContextMenu(string[] items, int minWidth, int maxHeight, byte font, Action<int> setIndex)
+        {
+            ResizePic background;
+            Add(background = new ResizePic(0x0BB8));
+            HoveredLabel[] labels = new HoveredLabel[items.Length];
+            int index = 0;
+
+            for (int i = 0; i < items.Length; i++)
+            {
+                string item = items[i];
+
+                if (item == null)
+                    item = string.Empty;
```

Comment:
Please wrap one-liners with curly braces, whenever you stumble upon them.

Reply:
heh, this was a copy pasta from ComboBox, normally VS would pick it up for me but i turned it off because i normally use tabs and i kept ending up with tabs everywhere. will fix.

Reply:
fixed.